File: solutions/2425.bitwise-xor-of-all-pairings.c
// @leet start
int xorAllNums(int* nums1, int nums1Size, int* nums2, int nums2Size) {
  int total = 0;
  if (nums2Size % 2 == 1) {
    for (int i = 0; i < nums1Size; i++) {
      total ^= nums1[i];
    }
  }

  if (nums1Size % 2 == 1) {
    for (int i = 0; i < nums2Size; i++) {
      total ^= nums2[i];
    }
  }

  return total;
}
// @leet end

File: solutions/2425.bitwise-xor-of-all-pairings.rs
// @leet start
impl Solution {
    pub fn xor_all_nums(nums1: Vec<i32>, nums2: Vec<i32>) -> i32 {
        let mut total_xor = 0;

        for (first, second) in [(&nums1, &nums2), (&nums2, &nums1)] {
            if second.len() % 2 == 1 {
                total_xor ^= first.iter().fold(0, |acc, e| acc ^ e);
            }
        }

        total_xor
    }
}
// @leet end

